source/host_resolver.c

```diff
@@ -123,8 +124,8 @@ struct host_entry {
        Let it tear on 32-bit systems, we don't care, we just want to see a change. This at least assumes cache coherency
        for the target architecture, which these days is a fairly safe assumption. Where it's not a safe assumption, we
        probably don't have multiple cores available anyways. */
-    volatile uint64_t last_use;
-    volatile bool keep_active;
+    struct aws_atomic_var last_use;
+    struct aws_atomic_var keep_active;
```

Comment:
Comment can be deleted, now that code is good

Reply:
This is trickier than I initially thought. `aws_atomic_var` gives a `size_t` so there are truncation issues with this change. I've reverted for now (so TSan will give a warning here). Long term, if the resolver only need to be aware of changes then this could be implemented with a 32-bit atomic counter that is incremented to avoid the problem of a 64-bit atomic value on a 32-bit machine. Or, alternatively, protecting `last_use` with a mutex.